### PR TITLE
Fixes infinite materials through recycling

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Fixes infinite materials from recycling
This PR should be merged on monday as it touches the very base of the game and i have not tested it at all , albeit it is very simple in design.

## Why It's Good For The Game
No more duping

## Testing
Not tested, merge for monday since i can't be trusted with even the most simple code-changes

## Changelog
:cl:
fix: Fixed infinite material duping exploit using smelters to get fractional sheets.
/:cl:

### CODE REVIEW

**Comment by Firefox13** [comment]
whoops wrong button

**Comment by MLGTASTICa** [comment]
> whoops wrong button

You took the bait,  unbelieveable

**Comment by Firefox13** [comment]
You literally said you didn't test it and that you can't be trusted.

**Comment by MLGTASTICa** [comment]
> You literally said you didn't test it and that you can't be trusted.

You shouldn't be putting  testing required and merge on monday tags on PR's whose code you can't read, any maintainer would realise these changes are so simple they don't need any testing at all.

**Comment by Firefox13** [comment]
If a person is unsure that their code works and asks to test it before merging it who am I to stop them?

**Comment by Firefox13** [comment]
*"If you didn't test it it doesn't work"*

**Comment by thevandie** [comment]
unironically, even if you're just adding an item or resprite, FUCKING TEST IT, for fucks sake

File: code/game/machinery/smelter.dm [hunk 1]
```diff
@@ -190,8 +190,14 @@
...
 
 	var/ejected_amount = min(initial(stack_type.max_amount), round(stored_material[material]), storage_capacity)
-	var/obj/item/stack/material/S = new stack_type(src, ejected_amount)
+	var/remainder = ejected_amount - round(ejected_amount)
+	var/obj/item/stack/material/S = new stack_type(src, round(ejected_amount))
+	var/shard
+	if(remainder)
+		shard = new /obj/item/material/shard(src, material, _amount = remainder)
 	eject(S, output_side)
+	if(shard)
```
Comment (Penwin0) [inline code/game/machinery/smelter.dm:199]: You can put both this and the definition of `var/shard` in the previous if statement, slightly tidier

**Review by Penwin0 — APPROVED** [review]
Should work fine

**Review by hyperioo — APPROVED** [review]

**Pull request merged**